src/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -24,6 +24,58 @@ private struct DateTimeParseData
             public byte OffsetToken;
         }
 
+        public static string FormatDateTimeOffset(DateTimeOffset value)
+        {
+            Span<byte> span = stackalloc byte[JsonConstants.MaximumFormatDateTimeOffsetLength];
+
+            JsonWriterHelper.WriteDateTimeOffsetTrimmed(span, value, out int bytesWritten);
+
+            return JsonReaderHelper.GetTextFromUtf8(span.Slice(0, bytesWritten));
```

Comment:
I'm confused why this is considered an improvement.  This is using the UTF8Formatter to format for "o" into a stackallocated span, and then transcoding that to a UTF16 string.  There's already an optimized path for "o" in coreclr that formats directly into a string.  I don't believe we should be adding extra code to all builds just to optimize the OOB implementation.

Reply:
The improvement isn't performance, it's the resulting format. As mentioned above, the ideal thing would be to a) change the "O" format to omit redundant details (breaking change?) or b) add a new format.

Because it's reusing all of `JsonText[Reader|Writer]`'s helpers, which always operates on `Span<byte>`, it needs to transcode.

Reply:
I see there's already a tracking issue (https://github.com/dotnet/corefx/issues/35976) for adding a new format.